### ISSUE TITLE
[Bug] AccountAddress & HashValue & PublicKey deserialize bug

### ISSUE THREAD

**Issue by jolestar** [libra/libra#3613]
# 🐛 Bug

AccountAddress & PublicKey custom deserialize implement has bug, can not deserialize from json value.

## To reproduce

**Code snippet to reproduce**
```rust
# Your code goes here
# Please make sure it does not require any external dependencies

#[test]
fn test_deserialize_from_json_value() {
    let address = AccountAddress::random();
    let json_value = serde_json::to_value(address).unwrap();
    let _address2: AccountAddress = serde_json::from_value(json_value).unwrap();
}

```

**Stack trace/error message**
```
// Paste the output here

called `Result::unwrap()` on an `Err` value: Error("invalid type: string \"09c516ab45cdbc7614b51506f9e2a7e7\", expected a borrowed string", line: 0, column: 0)
thread 'unit_tests::address_test::test_deserialize_from_json_value' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: string \"09c516ab45cdbc7614b51506f9e2a7e7\", expected a borrowed string", line: 0, column: 0)', types/src/unit_tests/address_test.rs:82:36
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1063
   5: std::io::Write::write_fmt
   6: std::io::impls::<impl std::io::Write for alloc::boxed::Box<W>>::write_fmt
             at src/libstd/io/impls.rs:156
   7: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   8: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   9: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
  10: std::panicking::default_hook
             at src/libstd/panicking.rs:221
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:470
  12: rust_begin_unwind
             at src/libstd/panicking.rs:378
  13: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  14: core::option::expect_none_failed
             at src/libcore/option.rs:1211
  15: core::result::Result<T,E>::unwrap
             at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libcore/result.rs:1003
  16: libra_types::unit_tests::address_test::test_deserialize_from_json_value
             at types/src/unit_tests/address_test.rs:82
  17: libra_types::unit_tests::address_test::test_deserialize_from_json_value::{{closure}}
             at types/src/unit_tests/address_test.rs:79

```

## Expected Behavior

The result generated by serde_json::to_value should be deserialize by serde_json::from_value.

## System information

**Please complete the following information:**
- 
- 
- 

## Additional context

The root cause of this error is：
```rust
          let s = <&str>::deserialize(deserializer)?;
          AccountAddress::from_str(s).map_err(D::Error::custom)
```
Use String to replace &str can fix this bug.

**Comment by Afiqnajmi79** [comment]
Have problem? Say at me can i finish you probleem

**Comment by jolestar** [comment]
Fixed at #3614
